UserInterface/GameManager/GameManager.cs

```diff
@@ -67,45 +92,72 @@ private void RunGame()
 
-                _pairManager.CheckPairForExistence(_pairs, _board, _animals, _children);
+                _deletionManager.RemoveInactiveItems(_gameItems);
+                Thread.Sleep(ConstantsRepository.ThreadSleep);
+                ConsoleKey? consoleKey = _userInterface.GetInputKey();
+            
+                _additionManager.ProcessAddNewItem(consoleKey, _gameItemsInfo, _gameItems, _board);    
 
-                _pairManager.CreatePair(_animals);
+                isGameOnGoing = (consoleKey != ConsoleKey.Q && consoleKey != ConsoleKey.Escape);
+            }
+        }
 
-                if (currentPair != null)
+        /// <summary>
+        /// Loads instances of the DLLs' children classes. 
+        /// </summary>
+        private void LoadInstancesOfDllChildrenClasses()
+        {
+            List<IItem> plugins = LoadPlugins().ToList();
+
+            foreach (var plugin in plugins)
+            {
+                GameItemsInfo info = new GameItemsInfo
                 {
-                    _pairs.Add(currentPair);
-                }
+                    Specie = plugin.Specie,
+                    Color = plugin.Color,
+                    Type = plugin.GetType()
+                };
 
-                _animalActions.AddChildren(_animals, _children);
-                _children.Clear();
+                _gameItemsInfo.Add(plugin.Letter, info);                
+            }
+        }
 
-                _animalActions.RemoveDeadAnimals(_animals);
-                Thread.Sleep(ConstantsRepository.ThreadSleep);
-                ConsoleKey? consoleKey = _userInterface.GetInputKey();
+        /// <summary>
+        /// Load DLL plugins from Plugins directory.
+        /// </summary>
+        /// <returns></returns>
+        private IEnumerable<IItem> LoadPlugins()
+        {
+            List<IItem> result = new List<IItem>();
 
-                switch (consoleKey)
-                {
-                    case ConsoleKey.A:
-                        _animalActions.AddAnimal(new Antilope(), _animals, _board);
-                        break;
-                    case ConsoleKey.L:
-                        _animalActions.AddAnimal(new Lion(), _animals, _board);
-                        break;
-                }
+            foreach (var dll in Directory.GetFiles($"{AppDomain.CurrentDomain.BaseDirectory}/{ConstantsRepository.PluginsFolder}", "*.dll"))
+            {
+                AssemblyLoadContext assemblyLoadContext = new AssemblyLoadContext(dll);
+                Assembly assembly = Assembly.LoadFrom(dll);
+                var plugins = assembly.ExportedTypes
+                    .Where(t => typeof(IItem).IsAssignableFrom(t)).ToList();
 
-                isGameOnGoing = (consoleKey != ConsoleKey.Q && consoleKey != ConsoleKey.Escape);
+                var plugin = plugins.Select(plugin => (IItem)Activator.CreateInstance(plugin));
+                result.Add((IItem)plugin);
```

Comment:
These three lines can be combined into one :
var plugins = assembly.ExportedTypes
    .Where(t => typeof(IItem).IsAssignableFrom(t))
    .Select(plugin => (IItem)Activator.CreateInstance(plugin));

And not sure if plugin is right variable name here maybe more like board items.

Reply:
Done.